app/models/Post.php

```diff
@@ -0,0 +1,127 @@
+<?php
+namespace app\models;
+class Post extends Entity {
+
+    /** Model Add a post
+     * @param $title
+     * @param $body
+     * @param $userId
+     * @return null
+     */
+    public function addPost($title, $body, $userId) {
```

Comment:
Extra challenge if you're interested: how could we refactor these add and update functions to allow extra columns to be added to the Post table?

Reply:
Further note - are the `addPost` and `updatePost` functions now redundant as you've added `create` and `update`.  If so, let's remove them

Reply:
For this one, I thought we can read the passed in array, and check whether all the attributes have corresponding columns, if not, alter the schema to add in the column first, then use a for loop and a placeholder to fill all the attributes in the query. Should be able to do the work.

Reply:
I have a workable solution, but I suspend that if I am trying to add in extra columns, for the existing data, the new column's value is null, there would be potential issues with that.

Reply:
Using an array is the option I'd consider.  In Magento save functions will often accept a key-pair array called `$data`